packages/flutter/test/widgets/fade_in_image_test.dart

```diff
@@ -156,7 +156,66 @@ Future<void> main() async {
       expect(findFadeInImage(tester).target.opacity, 1);
     });
 
-    testWidgets('shows a cached image immediately when skipFadeOnSynchronousLoad=true', (WidgetTester tester) async {
+    testWidgets('re-animates updated uncached image', (WidgetTester tester) async {
```

Comment:
This test appears to pass on master even without your change.

Reply:
Improved this test. Now it fails on master, as when the image is changed to an uncached image, the placeholder isn't shown until the animation starts.